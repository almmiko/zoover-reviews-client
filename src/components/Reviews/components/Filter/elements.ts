import styled from 'styled-components';

export const Title = styled.div`
  color: #757575;
  margin-bottom: 5px;
`;
