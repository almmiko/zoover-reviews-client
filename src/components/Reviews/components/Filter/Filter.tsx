import React from 'react';
import { Select } from 'antd';
import { Title } from './elements';
const { Option } = Select;

type Props = {
  onFilter: (value: string) => void,
}

class Filter extends React.Component<Props> {

  onChange = (value: string) => {
    this.props.onFilter(value)
  };

  render() {
    return (
      <div>
        <Title>Traveled with:</Title>
        <Select
          allowClear={true}
          style={{ width: 200 }}
          placeholder="Select type"
          optionFilterProp="children"
          onChange={this.onChange}
        >
          <Option value="FAMILY">FAMILY</Option>
          <Option value="FRIENDS">FRIENDS</Option>
          <Option value="OTHER">OTHER</Option>
          <Option value="COUPLE">COUPLE</Option>
          <Option value="SINGLE">SINGLE</Option>
        </Select>
      </div>

    );
  }
}

export default Filter;
