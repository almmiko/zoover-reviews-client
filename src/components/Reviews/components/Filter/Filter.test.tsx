import { mount, shallow } from 'enzyme';
import Filter from './Filter';
import React from 'react';
import { Select } from 'antd';

it('renders correctly', () => {
  const onFilter = jest.fn();
  const tree = shallow(<Filter onFilter={onFilter} />);
  expect(tree).toMatchSnapshot();
});

it('onFilter callback called with value', () => {
  const onFilter = jest.fn();
  const tree = mount(<Filter onFilter={onFilter} />);

  // @ts-ignore
  tree.find(Select).props().onChange('test');

  expect(onFilter).toHaveBeenLastCalledWith('test');
  expect(onFilter).toBeCalled();
});
