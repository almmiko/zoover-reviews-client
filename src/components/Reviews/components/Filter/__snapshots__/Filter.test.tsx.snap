// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <styled.div>
    Traveled with:
  </styled.div>
  <Select
    allowClear={true}
    choiceTransitionName="zoom"
    onChange={[Function]}
    optionFilterProp="children"
    placeholder="Select type"
    showSearch={false}
    style={
      Object {
        "width": 200,
      }
    }
    transitionName="slide-up"
  >
    <Option
      value="FAMILY"
    >
      FAMILY
    </Option>
    <Option
      value="FRIENDS"
    >
      FRIENDS
    </Option>
    <Option
      value="OTHER"
    >
      OTHER
    </Option>
    <Option
      value="COUPLE"
    >
      COUPLE
    </Option>
    <Option
      value="SINGLE"
    >
      SINGLE
    </Option>
  </Select>
</div>
`;
